tools/scylla-nodetool.cc

```diff
@@ -1578,6 +1624,58 @@ void stop_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
     client.post("/compaction_manager/stop_compaction", {{"type", compaction_type}});
 }
 
+void tablehistograms_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    const auto& [keyspace, table] = parse_keyspace_and_table(client, vm, "table");
+
+    auto get_estimated_histogram = [&client, &keyspace, &table] (std::string_view histogram) {
+        const auto res = client.get(format("/column_family/metrics/{}/{}:{}", histogram, keyspace, table));
+        const auto res_object = res.GetObject();
+        const auto& buckets_array = res["buckets"].GetArray();
+        const auto& bucket_offsets_array = res["bucket_offsets"].GetArray();
+
+        if (bucket_offsets_array.Size() + 1 != buckets_array.Size()) {
+            throw std::runtime_error(format("invalid estimated histogram {}, buckets must have one more element than bucket_offsets", histogram));
+        }
+
+        std::vector<int64_t> bucket_offsets, buckets;
+        for (size_t i = 0; i < bucket_offsets_array.Size(); ++i) {
+            buckets.emplace_back(buckets_array[i].GetInt64());
+            bucket_offsets.emplace_back(bucket_offsets_array[i].GetInt64());
+        }
+        buckets.emplace_back(buckets_array[buckets_array.Size() - 1].GetInt64());
+
+        return utils::estimated_histogram(std::move(bucket_offsets), std::move(buckets));
+    };
+
+    const auto row_size_hg = get_estimated_histogram("estimated_row_size_histogram");
+    const auto column_count_hg = get_estimated_histogram("estimated_column_count_histogram");
+    const auto read_latency_hg = buffer_samples::retrieve_from_api(client,
+            format("/column_family/metrics/read_latency/moving_average_histogram/{}:{}", keyspace, table));
+    const auto write_latency_hg = buffer_samples::retrieve_from_api(client,
+            format("/column_family/metrics/write_latency/moving_average_histogram/{}:{}", keyspace, table));
+    const auto sstables_per_read_hg = get_estimated_histogram("sstables_per_read_histogram");
+
+    fmt::print(std::cout, "{}/{} histograms\n", keyspace, table);
+    fmt::print(std::cout, "{:>10}{:>10}{:>18}{:>18}{:>18}{:>18}\n", "Percentile", "SSTables", "Write Latency", "Read Latency", "Partition Size", "Cell Count");
+    fmt::print(std::cout, "{:>10}{:>10}{:>18}{:>18}{:>18}{:>18}\n", "", " ", "(micros)", "(micros)", "(bytes)", "");
+    for (const auto percentile : {0.5, 0.75, 0.95, 0.98, 0.99}) {
+        fmt::print(
+                std::cout,
+                "{:<10}{:>10.2f}{:>18.2f}{:>18.2f}{:>18}{:>18}\n",
+                format("{}%", int(percentile * 100)),
```

Comment:
OK. but kind of weird as we are using `seastar::format()` along with `fmt::print()`, even if the latter would suffice. and the former is used for concatenating an `int`  and a `%`.

Reply:
unwrapped it in v2